src/route.rs

```diff
@@ -150,6 +292,15 @@ impl TryFrom<String> for RouteSpec {
     }
 }
 
+impl From<Vec<Segment>> for RouteSpec {
+    fn from(segments: Vec<Segment>) -> Self {
+        Self {
+            segments,
+            source: SmartString::new(),
```

Comment:
Is this not used for display and whatnot? I.e. should this reverse build this string?

Reply:
Fair, that makes sense